lib/widgets/product_detail.dart

```diff
@@ -58,11 +65,58 @@ class _ProductDetailState extends State<ProductDetail> {
         SizedBox(height: 8.0),
         _buildProductPrice(product),
         SizedBox(height: 8.0),
+        _buildFewItemsInStockMessage(product),
+        SizedBox(height: 8.0),
         _buildProductCategory(product.category),
       ],
     );
   }
 
+  Widget _buildFewItemsInStockMessage(Product product) {
+    int stockQuantityRemaining = 10;
+    if (widget.selectedItemSize != null && widget.selectedItemColor != null) {
+      product.stock.items.forEach((stockElement) {
+        if (stockElement.size == widget.selectedItemSize &&
+            stockElement.color == widget.selectedItemColor) {
+          stockQuantityRemaining = stockElement.quantity;
+        }
+      });
+    } else {
+      stockQuantityRemaining = product.stock.items.where((element)=>element.quantity>0).first.quantity;
+    }
+    List<Widget> textStructureWidgetList = List<Widget>();
+    if (stockQuantityRemaining < 10) {
+      textStructureWidgetList.add(Text(
+        "この商品の在庫は現在",
+        style: TextStyle(
+          color: paletteGreyColor,
+          fontSize: 12.0,
+          fontWeight: FontWeight.w600,
+        ),
+      ));
+      textStructureWidgetList.add(Text(
+        stockQuantityRemaining.toString() + "個",
+        style: TextStyle(
+          color: paletteForegroundColor,
+          fontSize: 12.0,
+          fontWeight: FontWeight.w600,
+        ),
+      ));
+      textStructureWidgetList.add(Text(
+        "です",
+        style: TextStyle(
+          color: paletteGreyColor,
+          fontSize: 12.0,
+          fontWeight: FontWeight.w600,
+        ),
+      ));
+    }
+    return Row(
```

Comment:
Instead of returning an empty `Row` when `stockQuantityRemaining >= 10`, you can do:

```
return stockQuantityRemaining < 10 ? Row(
  mainAxisAlignment: MainAxisAlignment.end,
  children: textStructureWidgetList,
) : Container();
```

Reply:
nice tip, I didn't know this was possible